go-controller/observability-lib/sampledecoder/sample_decoder.go

```diff
@@ -133,90 +135,88 @@ func (d *SampleDecoder) DecodeCookieIDs(obsDomainID, obsPointID uint32) (string,
-		return "", fmt.Errorf("failed to get message for db object %v", dbObj)
+	var event model.NetworkEvent
+	switch o := dbObj.(type) {
+	case *nbdb.ACL:
+		event = newACLEvent(o)
+	}
+	if event == nil {
+		return nil, fmt.Errorf("failed to build network event for db object %v", dbObj)
 	}
-	return msg, nil
+	return event, nil
 }
 
-func getMessage(dbObj interface{}) string {
-	switch o := dbObj.(type) {
-	case *nbdb.ACL:
-		var action string
-		switch o.Action {
-		case nbdb.ACLActionAllow, nbdb.ACLActionAllowRelated, nbdb.ACLActionAllowStateless:
-			action = "Allowed"
-		case nbdb.ACLActionDrop:
-			action = "Dropped"
-		case nbdb.ACLActionPass:
-			action = "Delegated to network policy"
-		default:
-			action = "Action " + o.Action
-		}
-		actor := o.ExternalIDs[libovsdbops.OwnerTypeKey.String()]
-		var msg string
-		switch actor {
-		case libovsdbops.AdminNetworkPolicyOwnerType:
-			msg = fmt.Sprintf("admin network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.BaselineAdminNetworkPolicyOwnerType:
-			msg = fmt.Sprintf("baseline admin network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.MulticastNamespaceOwnerType:
-			msg = fmt.Sprintf("multicast in namespace %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.MulticastClusterOwnerType:
-			msg = fmt.Sprintf("cluster multicast policy, direction %s", o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.NetpolNodeOwnerType:
-			msg = "default allow from local node policy, direction ingress"
-		case libovsdbops.NetworkPolicyOwnerType:
-			msg = fmt.Sprintf("network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.NetpolNamespaceOwnerType:
-			msg = fmt.Sprintf("network policies isolation in namespace %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.EgressFirewallOwnerType:
-			msg = fmt.Sprintf("egress firewall in namespace %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()])
-		case libovsdbops.UDNIsolationOwnerType:
-			msg = fmt.Sprintf("UDN isolation of type %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()])
+func newACLEvent(o *nbdb.ACL) *model.ACLEvent {
+	actor := o.ExternalIDs[libovsdbops.OwnerTypeKey.String()]
+	event := model.ACLEvent{
+		Action: o.Action,
+		Actor:  actor,
+	}
+	switch actor {
+	case libovsdbops.NetworkPolicyOwnerType:
+		nsname := strings.SplitN(o.ExternalIDs[libovsdbops.ObjectNameKey.String()], ":", 2)
+		if len(nsname) == 2 {
+			event.Namespace = nsname[0]
+			event.Name = nsname[1]
+		} else {
+			// do not split (which use cases?)
```

Comment:
this should never happen, you can return an error is it does

Reply:
done in https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4818/commits/b8ab657385ff1c6c21c80ed998cbb6cdaa59af3b